fili-core/src/test/java/com/yahoo/bard/webservice/data/config/dimension/TestLookupDimensionConfig.java

```diff
@@ -13,37 +18,88 @@
  * Hold all of the Configuration information for the look up dimensions.
  */
 public class TestLookupDimensionConfig extends TestDimensionConfig implements LookupDimensionConfig {
-    private List<String> namespaces;
+    private final List<String> namespaces;
 
     /**
      * Constructor.
      *
      * @param dimensionConfig Configuration properties for dimensions
      * @param namespaces List of namespaces used for Lookup
      */
-    public TestLookupDimensionConfig (@NotNull DimensionConfig dimensionConfig, List<String> namespaces) {
-        super(
+    public TestLookupDimensionConfig(@NotNull DimensionConfig dimensionConfig, List<String> namespaces) {
+        this(
                 TestApiDimensionName.valueOf(dimensionConfig.getApiName().toUpperCase(Locale.ENGLISH)),
                 dimensionConfig.getPhysicalName(),
                 dimensionConfig.getKeyValueStore(),
                 dimensionConfig.getSearchProvider(),
                 dimensionConfig.getFields(),
-                dimensionConfig.getDefaultDimensionFields()
+                dimensionConfig.getDefaultDimensionFields(),
+                namespaces
         );
-        this.setNamespaces(namespaces);
     }
 
     /**
-     * Sets the namespace chain.
+     * Constructor.
      *
-     * @param namespaces The list of namespaces
+     * @param apiName  API Name of the dimension
+     * @param physicalName Physical name of the dimension
+     * @param keyValueStore  KeyValueStore for the dimension
+     * @param searchProvider  SearchProvider for the dimension
+     * @param fields  Fields of the dimension
+     * @param defaultFields  Default fields of the dimension
+     * @param namespaces List of namespaces used for Lookup
      */
-    private void setNamespaces(List<String> namespaces) {
-        this.namespaces = namespaces;
+    public TestLookupDimensionConfig(
+            TestApiDimensionName apiName,
+            String physicalName,
+            KeyValueStore keyValueStore,
+            SearchProvider searchProvider,
+            LinkedHashSet<DimensionField> fields,
+            LinkedHashSet<DimensionField> defaultFields,
+            List<String> namespaces
+    ) {
+        super(apiName, physicalName, keyValueStore, searchProvider, fields, defaultFields);
+        this.namespaces = Collections.unmodifiableList(namespaces);
```

Comment:
Generally, we're OK letting test-code be modifiable. This is a good impulse, but not for test code 😉